### PR TITLE
Improve floating-point performance

### PR DESCRIPTION
**Pull request by Theodor-Lindberg**
Use `cast_mantissa` for multiplication and the power function to improve performance.

### CODE REVIEW

File: src/apyfloat.cc [hunk 1]
```diff
@@ -943,38 +946,30 @@ APyFloat APyFloat::operator*(const APyFloat& y) const
...
-            if (new_man >= two) {
-                tmp_exp++;
-            } else {
-                // Align with longer result
-                new_man <<= 1;
-            }
+        auto new_man_bits = sum_man_bits;
+
+        // Carry from multiplication
+        if (new_man & (one << 1)) {
```
Comment (oscargus) [inline src/apyfloat.cc:952]:
```suggestion
        if (new_man & two) {
```
?

**Comment by Theodor-Lindberg** [comment]
There was a slight degradation in performance for other operations, except multiplication, when moving the overflow check in `_cast`, commit 44fb72b. This is currently being investigated. Commit 96b665c helped but hasn't restored the benchmarks fully.

**Comment by oscargus** [comment]
My guess is that the performance is a bit "random". I made a change recently to compute a thing once and store in a variable, rather than twice. That halved the performance and also affected unrelated things.

**Comment by oscargus** [comment]
I am quite confident that it makes sense to make sure that things are align identically (as before), especially when moving to arrays it will save computations.

**Comment by Theodor-Lindberg** [comment]
> I am quite confident that it makes sense to make sure that things are align identically (as before), especially when moving to arrays it will save computations.

Okay, so just undo those two commits? If I understand you right.

**Pull request merged**